### PR TITLE
Leaderboard improvements

### PR DESCRIPTION
**Pull request by BlackYps**
Adds the ability to select a different season than the currently active one.
Also fixes the sorting of the leaderboard tabs

### CODE REVIEW

File: src/main/java/com/faforever/client/leaderboard/LeaderboardService.java [hunk 1]
```diff
@@ -102,26 +102,19 @@ public Flux<LeagueSeason> getActiveSeasons() {
...
-                                                                                                                       false))
+                                                                                                                   .eq(league.id()))
                                                                                                      .addSortingRule(
                                                                                                          "startDate",
                                                                                                          false);
-    return fafApiAccessor.getMany(navigator)
-                         .next().map(leaderboardMapper::map);
+    return fafApiAccessor.getMany(navigator).map(leaderboardMapper::map)
+                         .cache();
   }
```
Comment (Sheikah45) [inline src/main/java/com/faforever/client/leaderboard/LeaderboardService.java:118]: The only thing is that this list could get very long over time. So we may want some kind of pagination

Comment (BlackYps) [reply src/main/java/com/faforever/client/leaderboard/LeaderboardService.java:118]: If we want to access a season that is not in the first page we would have to be able to load earlier pages. That sounds very cumbersome from a ui perspective. I'm not sure how we could reflect this pagination in the client ui

Comment (BlackYps) [reply src/main/java/com/faforever/client/leaderboard/LeaderboardService.java:118]: besides, you are technically correct that this list could get long, but we have four seasons per year, so in ten years we will be at a total of around 50. I think this is still small enough to load it in one go

Comment (Sheikah45) [reply src/main/java/com/faforever/client/leaderboard/LeaderboardService.java:118]: It was less the loading I was thinking about and more how long the list in the UI could be and how that would be to navigate

Comment (BlackYps) [reply src/main/java/com/faforever/client/leaderboard/LeaderboardService.java:118]: It seems the combox only shows at max 10 entries and creates a scrollbar automatically if there are more

**Pull request merged**